.github/workflows/swift-toolchain.yml

```diff
@@ -73,7 +73,19 @@ jobs:
           sudo DEBIAN_FRONTEND=noninteractive apt-get -qq update -yq
           sudo DEBIAN_FRONTEND=noninteractive apt-get -qq -o Dpkg::Use-Pty=0 install -yq repo libxml2-utils
 
-          repo init --quiet --groups default --depth 1 -u https://github.com/compnerd/swift-build -b main
+          # Which branch is this workflow based on
+          branch_version_string=${{ github.event.inputs.swift_version || '0.0.0' }}
+          if echo "$branch_version_string" | grep '\.'; then
+            branch_version="$(echo "$branch_version_string" | cut -d '.' -f 1).$(echo "$branch_version_string" | cut -d '.' -f 2)"
```

Comment:
Hmm, what do you think of:

`echo ${branch_version_string} | awk -F. '{ print $1"."$2 }'` as an alternative?

Reply:
it needs a bit of extra care when version is a single number hmm